src/test/java/uk/gov/ons/ctp/integration/envoycuc/mockclient/MockClient.java

```diff
@@ -0,0 +1,292 @@
+  }
+
+  private void postRequest(List<String> requestKeyList, final RateLimiterClientRequest request) {
+    for (String requestKey : requestKeyList) {
+      Map<String, List<Integer>> postedMap = postingsTimeMap.get(requestKey);
+      if (postedMap == null) {
+        return;
+      }
+      final String[] keyConstituents = requestKey.split("-");
+      final String keyType = keyConstituents[keyConstituents.length - 1];
+      String postingsListKey = null;
+      if (keyType.equals("UPRN")) {
+        postingsListKey = request.getUprn().getValue() + "";
+      }
+      if (keyType.equals("IP")) {
+        postingsListKey = request.getIpAddress();
+      }
+      if (keyType.equals("TELNO")) {
+        postingsListKey = request.getTelNo();
+      }
+      postedMap.computeIfAbsent(postingsListKey, k -> new ArrayList<>());
+      final Date now = new Date(System.currentTimeMillis());
+      SimpleDateFormat dmformatter = new SimpleDateFormat("DDDHH");
+      int dayHour = Integer.parseInt(dmformatter.format(now));
+      postedMap.get(postingsListKey).add(dayHour);
+    }
+  }
+
+  @PostConstruct
+  private void setupAllowances() {
+    allowanceMap.put("UAC-FALSE-SMS-HH-UPRN", 5);
+    allowanceMap.put("UAC-FALSE-SMS-SPG-UPRN", 5);
+    allowanceMap.put("UAC-FALSE-SMS-CE-UPRN", 5);
+    allowanceMap.put("UAC-FALSE-SMS-HH-TELNO", 10);
+    allowanceMap.put("UAC-FALSE-SMS-SPG-TELNO", 10);
+    allowanceMap.put("UAC-FALSE-SMS-CE-TELNO", 5);
+    allowanceMap.put("UAC-FALSE-SMS-HH-IP", 100);
+    allowanceMap.put("UAC-FALSE-SMS-SPG-IP", 100);
+    allowanceMap.put("UAC-FALSE-SMS-CE-IP", 100);
+    allowanceMap.put("UAC-TRUE-SMS-HH-UPRN", 10);
+    allowanceMap.put("UAC-TRUE-SMS-SPG-UPRN", 10);
+    allowanceMap.put("UAC-TRUE-SMS-CE-UPRN", 50);
+    allowanceMap.put("UAC-TRUE-SMS-HH-TELNO", 10);
+    allowanceMap.put("UAC-TRUE-SMS-SPG-TELNO", 10);
+    allowanceMap.put("UAC-TRUE-SMS-CE-TELNO", 50);
+    allowanceMap.put("UAC-TRUE-SMS-HH-IP", 100);
+    allowanceMap.put("UAC-TRUE-SMS-SPG-IP", 100);
+    allowanceMap.put("UAC-TRUE-SMS-CE-IP", 100);
+    allowanceMap.put("UAC-FALSE-POST-HH-UPRN", 1);
+    allowanceMap.put("UAC-FALSE-POST-SPG-UPRN", 1);
+    allowanceMap.put("UAC-FALSE-POST-CE-UPRN", 1);
+    allowanceMap.put("UAC-TRUE-POST-HH-UPRN", 5);
+    allowanceMap.put("UAC-TRUE-POST-SPG-UPRN", 5);
+    allowanceMap.put("UAC-TRUE-POST-CE-UPRN", 50);
+    allowanceMap.put("QUESTIONNAIRE-FALSE-POST-HH-UPRN", 1);
+    allowanceMap.put("QUESTIONNAIRE-FALSE-POST-SPG-UPRN", 1);
+    allowanceMap.put("QUESTIONNAIRE-TRUE-POST-HH-UPRN", 5);
+    allowanceMap.put("QUESTIONNAIRE-TRUE-POST-SPG-UPRN", 5);
+    allowanceMap.put("QUESTIONNAIRE-TRUE-POST-CE-UPRN", 50);
+    allowanceMap.put("CONTINUATION-FALSE-POST-HH-UPRN", 12);
+    allowanceMap.put("CONTINUATION-FALSE-POST-SPG-UPRN", 12);
+  }
+
+  @PostConstruct
+  private void setupTimeMaps() {
+    postingsTimeMap.put("UAC-FALSE-SMS-HH-UPRN", getNewTimeMap());
```

Comment:
no need to change, but you might have considered one map combining both `allowanceMap` and `postingsTimeMap`.  Then no chance of missing a scenario.

Reply:
Taken casre of this in a slightly different way - creates the second map from the first